numpy-stubs/__init__.pyi

```diff
@@ -792,3 +797,305 @@ class AxisError(ValueError, IndexError):
+)
+
+def take(
+    a: _ArrayLike,
+    indices: _ArrayLikeInt,
+    axis: Optional[int] = ...,
+    out: Optional[ndarray] = ...,
+    mode: _Mode = ...,
+) -> ndarray: ...
+def reshape(a: _ArrayLike, newshape: _ShapeLike, order: _Order = ...) -> ndarray: ...
+def choose(
+    a: _ArrayLike,
+    choices: Sequence[_ArrayLike],
+    out: Optional[ndarray] = ...,
+    mode: _Mode = ...,
+) -> ndarray: ...
+def repeat(
+    a: _ArrayLike, repeats: _ArrayLikeInt, axis: Optional[int] = ...
+) -> ndarray: ...
+def put(a: ndarray, ind: _ArrayLikeInt, v: _ArrayLike, mode: _Mode = ...) -> None: ...
+def swapaxes(a: _ArrayLike, axis1: int, axis2: int) -> ndarray: ...
+def transpose(a: _ArrayLike, axes: Optional[Sequence[int]] = ...) -> ndarray: ...
+def partition(
+    a: _ArrayLike,
+    kth: _ArrayLikeInt,
+    axis: Optional[int] = ...,
+    kind: _PartitionKind = ...,
+    order: Union[None, str, Sequence[str]] = ...,
+) -> ndarray: ...
+def argpartition(
+    a: _ArrayLike,
+    kth: _ArrayLikeInt,
+    axis: Optional[int],
+    kind: _PartitionKind = ...,
+    order: Union[None, str, Sequence[str]] = ...,
+) -> ndarray: ...
+def sort(
+    a: _ArrayLike,
+    axis: Optional[int] = ...,
+    kind: Optional[_SortKind] = ...,
+    order: Union[None, str, Sequence[str]] = ...,
+) -> ndarray: ...
+def argsort(
+    a: _ArrayLike,
+    axis: Optional[int] = ...,
+    kind: Optional[_SortKind] = ...,
+    order: Union[None, str, Sequence[str]] = ...,
+) -> ndarray: ...
+@overload
+def argmax(a, axis: None = ..., out: Optional[ndarray] = ...) -> integer: ...
+@overload
+def argmax(
+    a, axis: int = ..., out: Optional[ndarray] = ...
+) -> Union[integer, ndarray]: ...
+@overload
+def argmin(a, axis: None = ..., out: Optional[ndarray] = ...) -> integer: ...
+@overload
+def argmin(
+    a, axis: int = ..., out: Optional[ndarray] = ...
+) -> Union[integer, ndarray]: ...
+def searchsorted(
+    a: _ArrayLike,
+    v: _ArrayLike,
+    side: _Side = ...,
+    sorter: Optional[_ArrayLikeInt] = ...,
+) -> ndarray: ...
+def resize(a: _ArrayLike, new_shape: _ShapeLike) -> ndarray: ...
+def squeeze(a: _ArrayLike, axis: Optional[_ShapeLike] = ...) -> ndarray: ...
+def diagonal(
+    a: _ArrayLike, offset: int = ..., axis1: int = ..., axis2: int = ...
+) -> ndarray: ...
+def trace(
+    a: _ArrayLike,
+    offset: int = ...,
+    axis1: int = ...,
+    axis2: int = ...,
+    dtype: _DtypeLike = ...,
+    out: Optional[ndarray] = ...,
+) -> Union[number, ndarray]: ...
+def ravel(a: _ArrayLike, order: _Order = ...) -> ndarray: ...
+def nonzero(a: _ArrayLike) -> Tuple[ndarray, ...]: ...
+def shape(a: _ArrayLike) -> _Shape: ...
+def compress(
+    condition: _ArrayLikeBool,
+    a: _ArrayLike,
+    axis: Optional[int] = ...,
+    out: Optional[ndarray] = ...,
+) -> ndarray: ...
+def clip(
+    a: _ArrayLike,
+    a_min: _ArrayLike,
+    a_max: _ArrayLike,
+    out: Optional[ndarray] = ...,
+    **kwargs: Any,
+) -> Union[number, ndarray]: ...
+@overload
```

Comment:
In principle there are also (at least) two other overloads for ``sum()`` and many other function:
* *a* is a ``np.number``: the returned value is a ``TypeVar`` with respect to *a*.
* *a* is a ``_Scalar``: the returned value is a ``np.number`` but not necasarilly a proper ``TypeVar``, as *a* can also be a builtin type.

Reply:
Overloads implemented in https://github.com/numpy/numpy-stubs/pull/59/commits/ea31239f1c523111d39946d6cef834f6898e4333.